packages/loot-core/src/shared/schedules.ts

```diff
@@ -401,22 +401,22 @@ export function describeSchedule(schedule, payee) {
   }
 }
 
-export function getUpcomingDays(upcomingLength = '7'): number {
-  const today = monthUtils.currentDay();
+export function getUpcomingDays(
+  upcomingLength = '7',
+  today = monthUtils.currentDay(), // for testability
+): number {
   const month = monthUtils.getMonth(today);
 
   switch (upcomingLength) {
     case 'currentMonth': {
       const day = monthUtils.getDay(today);
       const end = monthUtils.getDay(monthUtils.getMonthEnd(today));
-      return end - day + 1;
+      return end - day;
     }
     case 'oneMonth': {
-      return (
-        monthUtils.differenceInCalendarDays(
-          monthUtils.nextMonth(month),
-          month,
-        ) + 1
+      return monthUtils.differenceInCalendarDays(
+        monthUtils.nextMonth(month),
+        month,
```

Comment:
In that case...

```suggestion
      return (
        monthUtils.differenceInCalendarDays(
          monthUtils.nextMonth(month),
          month,
        ) - 1
```

The +1 was to make it run to the same day in the next month, my earlier patch fixes that so that the +1 isn't needed for that behaviour, the -1 will mean it runs for one less day than the amount of days in the month which will effectively "include" today in the calculation.